lib/viewport.dart

```diff
@@ -74,7 +74,9 @@ class Viewport extends ViewportTransform {
       final focus = size.y / 2 - temp.y;
 
       if (focus.abs() > margin) {
-        y = size.y / 2 + (position.y * scale) + (focus < 0 ? margin : -margin);
+        y = size.y / 2 +
+            (yFlip ? -1 : 1) *
+                ((position.y * scale) + (focus < 0 ? margin : -margin));
```

Comment:
I wonder if it would be worth to start to split this expression in named variables, it is starting to be quite big and that would make it easier to read